Source/DafnyLanguageServer/Workspace/Compilation.cs

```diff
@@ -0,0 +1,461 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Subjects;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Util;
+using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer.Workspace;
+
+public delegate Compilation CreateCompilation(
+  DafnyOptions options,
+  ExecutionEngine boogieEngine,
+  CompilationInput compilation);
+
+/// <summary>
+/// The compilation of a single document version.
```

Comment:
How to differentiate this compilation from the compilation process when translating to target files?
It might help to use a different word. You could use
* `Verification` (more accurate given the end result)
* `VerificationPipeline`
* `CheckingPhases`
* `VerificationProcess`

Reply:
This class will also be used for the CLI, and also do translation to other languages